### PR TITLE
Removing version check for python2 from files

### PR DESCRIPTION
**Pull request by deepaklorkhatri007**
This pull request addresses issue #2490

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Tests/test_SeqIO_SeqXML.py [hunk 1]
```diff
@@ -52,9 +52,7 @@ def test_special_characters_desc(self):
             self.records["dna"][2].description,
             'some special characters in the description\n<tag> "quoted string"'
         )
-
-    # TODO - Fix this failure under Windows with Python 3.1 and 3.2
-    if not (sys.platform == "win32" and sys.version_info[0] >= 3):
+    if not sys.platform == "win32":
```
Comment (mdehoon) [inline Tests/test_SeqIO_SeqXML.py]: Do we still need this? Maybe this is a good time to find out.

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: You mean remove it, and see if AppVeyor breaks? Or do either of you have access to a Windows machine or two for testing?

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: If we do keep this if-statement, it would be nicer as:

```python
if sys.platform != "win32":
```

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: yeah, I don't know about this one that's why in [this comment](https://github.com/biopython/biopython/issues/2490#issuecomment-571080136) I said that it is a special one.

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: I am testing this on a Windows machine. will let you know the results in a while :)

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: Wait the code is
 ```
if the operating system is not windows:
    add this...
``` 
why do we have to check this on windows?

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: (One or both of us is confused - easily done with multiple negatives - I think I've got this straight in my head now)

My reading of the code is we include ``test_unicode_characters_desc`` on Linux or Mac (and so this is being run on Linux via TravisCI).

Currently on Windows that test method ``test_unicode_characters_desc`` does _not_ get defined, and so is not run (i.e. not run on Windows via AppVeyor).

What happens if we _always_ include ``test_unicode_characters_desc`` (even on Windows)?

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: I have run the test on Linux without the function and nothing failed so it is of no use in Linux and Mac. It is not being used with windows before so windows don't need this. So the question is do we really need this function?

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: I think you have this backwards. Removing the ``test_unicode_characters_desc`` should always work on any machine.

We want to know if always _including_ ``test_unicode_characters_desc`` works.

Currently (I think) ``test_unicode_characters_desc`` is not being used on Windows. The question is if you add this test on Windows, does it pass?

Comment (MarkusPiotrowski) [reply Tests/test_SeqIO_SeqXML.py]: Why should there be a failure when you remove a test?

The test is testing if we are able to read certain unicode characters correctly. This didn't work on Windows with Python 3.1 and 3.2, so it was skipped on Windows. Now it is probably working fine on Windows. So try removing the ìf` clause to run this test aslo on Windows.

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: ohh, my bad! now I got this. I am really very sorry for the misunderstanding there 🙈

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: Don't worry - double negatives are horrible, we all struggle with them.

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: @peterjc `python setup.py build` is failing on both Linux and Windows due to the deletion of Bio/_py3k.
```
running build
running build_py
copying Bio/Seq.py -> build/lib.linux-x86_64-3.7/Bio
copying Bio/SeqRecord.py -> build/lib.linux-x86_64-3.7/Bio
copying Bio/Entrez/__init__.py -> build/lib.linux-x86_64-3.7/Bio/Entrez
copying Bio/ExPASy/__init__.py -> build/lib.linux-x86_64-3.7/Bio/ExPASy
copying Bio/KEGG/REST.py -> build/lib.linux-x86_64-3.7/Bio/KEGG
copying Bio/SeqIO/__init__.py -> build/lib.linux-x86_64-3.7/Bio/SeqIO
copying Bio/SeqIO/Interfaces.py -> build/lib.linux-x86_64-3.7/Bio/SeqIO
copying Bio/TogoWS/__init__.py -> build/lib.linux-x86_64-3.7/Bio/TogoWS
error: package directory 'Bio/_py3k' does not exist
```

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: Before editing anything i was getting this error
```
  File "C:\Users\dkmax\Github\biopython\Tests\test_MafIO_index.py", line 590, in test_correct_retrieval_1
    3227479), 1)
  File "C:\Users\dkmax\Github\biopython\build\lib.win-amd64-3.7\Bio\AlignIO\MafIO.py", line 656, in get_spliced
    fetched = list(self.search(starts, ends))
  File "C:\Users\dkmax\Github\biopython\build\lib.win-amd64-3.7\Bio\AlignIO\MafIO.py", line 630, in search
    % (rec_start, rec_end, offset, start, end)
ValueError: Expected 3185728-3185797 @ offset 1974167, found 3185634-3185640

----------------------------------------------------------------------
Ran 520 tests in 317.877 seconds

FAILED (failures = 1)
```

After removing the if clause from `test_SeqIO_SeqXML.py` i am getting this error
```
======================================================================
ERROR: test_correct_retrieval_1 (test_MafIO_index.TestSpliceGoodMAF)
Correct retrieval of Cnksr3 in mouse.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\dkmax\Github\biopython\Tests\test_MafIO_index.py", line 590, in test_correct_retrieval_1
    3227479), 1)
  File "C:\Users\dkmax\Github\biopython\build\lib.win-amd64-3.7\Bio\AlignIO\MafIO.py", line 656, in get_spliced
    fetched = list(self.search(starts, ends))
  File "C:\Users\dkmax\Github\biopython\build\lib.win-amd64-3.7\Bio\AlignIO\MafIO.py", line 630, in search
    % (rec_start, rec_end, offset, start, end)
Test special unicode characters in the description.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\dkmax\Github\biopython\Tests\test_SeqIO_SeqXML.py", line 58, in test_unicode_characters_desc
    self.assertEqual(self.records["rna"][2].description, "åÅüöÖßøä¢£$€香肠")
AssertionError: 'Ã¥Ã…Ã¼Ã¶Ã–ÃŸÃ¸Ã¤Â¢Â£$â‚¬é¦™è‚\xa0' != 'åÅüöÖßøä¢£$€香肠'
- Ã¥Ã…Ã¼Ã¶Ã–ÃŸÃ¸Ã¤Â¢Â£$â‚¬é¦™è‚ 
+ åÅüöÖßøä¢£$€香肠


----------------------------------------------------------------------
Ran 520 tests in 302.114 seconds

FAILED (failures = 2)
```

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: I have included `if sys.platform != "win32":` in test_SeqIO_SeqXML.py and removed `Bio.py3k` from setup.py :)

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: Oops - 18294c7d2bf06270cba174c1637ab761e33d9a5a fixed that. Sorry!

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: OK, so it looks like some Windows weirdness is still there, even on Python 3. That might deserve a followup issue, but ``if sys.platform != "win32":`` is enough for this pull request.

File: Tests/test_SeqIO_SeqXML.py [hunk 2]
```diff
@@ -52,9 +52,7 @@ def test_special_characters_desc(self):
             self.records["dna"][2].description,
             'some special characters in the description\n<tag> "quoted string"'
         )
-
-    # TODO - Fix this failure under Windows with Python 3.1 and 3.2
-    if not (sys.platform == "win32" and sys.version_info[0] >= 3):
+    if sys.platform != "win32":
```
Comment (peterjc) [inline Tests/test_SeqIO_SeqXML.py]: I'd have expected flake8 to say there should still be a blank line here...

**Comment by peterjc** [comment]
Thank you!

File: Tests/test_SeqIO_SeqXML.py (hunk 1 above)
Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: @peterjc can you give me some hints for solving those weird errors I am getting in windows? I want to give help for windows compatibility also.

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: The original workaround was added 8 years ago:
https://github.com/biopython/biopython/commit/0fab81a6b32e8b8a8c3ea8d7226e2f9603686d23

I don't immediately recall any more details - hopefully I know a little more about encodings than I did then.

Comment (deepaklorkhatri007) [reply Tests/test_SeqIO_SeqXML.py]: will try solving the problems thanks for the lead 😇

Comment (peterjc) [reply Tests/test_SeqIO_SeqXML.py]: It caught @mdehoon's eye - turns out to be similar to something he worked on recently in the Entrez parser - see pull request #2525

**Pull request merged**